lib/vmselectapi/server.go

```diff
@@ -931,8 +937,13 @@ func (s *Server) processSeriesSearch(ctx *vmselectRequestCtx) error {
 	if err := bi.Error(); err != nil {
 		return fmt.Errorf("search error: %w", err)
 	}
-	ctx.qt.Printf("sent %d blocks to vmselect", blocksRead)
 
+	s.seriesPerQueryRead.Update(float64(len(seriesStats)))
+	for _, v := range seriesStats {
+		s.samplesPerSeriesRead.Update(float64(v))
```

Comment:
This histogram shows the number of raw samples per series per each `vmstorage`. This metric cannot help estimating memory usage neither at `vmstorage` nor at `vmselect`, since `vmstorage` streams blocks with raw samples directly from persistent storage to `vmselect` - it doesn't unpack and hold these blocks in memory. This metric must be created at `vmselect` level.